modules/serviceregistry/src/test/java/org/opencastproject/serviceregistry/impl/ServiceRegistryJpaImplTest.java

```diff
@@ -252,69 +334,89 @@ public void testCancelUndispatchablesOrphanedByActivatingNode() throws Exception
+    launchDispatcherOnce(false);
     try {
-      barrier.waitForJobs(2000);
-      Assert.fail();
+      barrier.waitForJobs(JOB_BARRIER_TIMEOUT);
+      Assert.fail("Did not recieve a timeout exception");
     } catch (Exception e) {
       Assert.assertEquals(0, serviceRegistryJpaImpl.dispatchPriorityList.size());
+    } finally {
+      logger.debug("end testHostsBeingRemovedFromPriorityList");
+    }
+  }
+
+  @Test
+  public void testIgnoreHostsInPriorityList() throws Exception {
+    Job testJob = serviceRegistryJpaImpl.createJob(TEST_HOST, TEST_SERVICE_2, TEST_OPERATION, null, null, true, null);
+    Job testJob2 = serviceRegistryJpaImpl.createJob(TEST_HOST, TEST_SERVICE, TEST_OPERATION, null, null, true, null);
+    serviceRegistryJpaImpl.dispatchPriorityList.put(testJob2.getId(), TEST_HOST);
+    JobBarrier barrier = new JobBarrier(null, serviceRegistryJpaImpl, testJob, testJob2);
+    launchDispatcherOnce(false);
+    try {
+      barrier.waitForJobs(JOB_BARRIER_TIMEOUT);
+      Assert.fail("Did not recieve a timeout exception");
+    } catch (Exception e) {
+      logger.debug("job1: '{}'", serviceRegistryJpaImpl.getJob(testJob.getId()));
+      logger.debug("job2: '{}'", serviceRegistryJpaImpl.getJob(testJob2.getId()));
+      for (Long jobId :serviceRegistryJpaImpl.dispatchPriorityList.keySet()) {
+        logger.debug("job in priority queue: {}, {}", jobId, serviceRegistryJpaImpl.dispatchPriorityList.get(jobId));
+      }
+      // Mock http client always returns 503 for this path so it won't be dispatched anyway
+      testJob = serviceRegistryJpaImpl.getJob(testJob.getId());
+      Assert.assertTrue("First job should not have a processing host", StringUtils.isBlank(testJob.getProcessingHost()));
+      Assert.assertEquals("First job is queueued", Job.Status.QUEUED, testJob.getStatus());
+
+      // Mock http client always returns 204 for this path, but it should not be dispatched
+      // because the host is in the dispatchPriorityList
+      testJob2 = serviceRegistryJpaImpl.getJob(testJob2.getId());
+      Assert.assertTrue("Second job should not have a processing host", StringUtils.isBlank(testJob2.getProcessingHost()));
+      Assert.assertEquals("Second job is queueued", Job.Status.QUEUED, testJob2.getStatus());
+
+      Assert.assertEquals(1, serviceRegistryJpaImpl.dispatchPriorityList.size());
+      String blockingHost = serviceRegistryJpaImpl.dispatchPriorityList.get(testJob2.getId());
+      Assert.assertEquals(TEST_HOST, blockingHost);
+    } finally {
+      logger.debug("end testIgnoreHostsInPriorityList");
     }
   }
 
   private void assertHostloads(Job j, Float a, Float b, Float c) throws Exception {
-    Thread.sleep(1100); //1100 is 100ms more than the minimum job dispatch interval.  Setting this lower causes race conditions.
+    // launch the  dispatcher and wait a little longer for dispatch to complete before getting job
+    launchDispatcherOnce(false);
+    Thread.sleep(3 * DISPATCH_START_DELAY);
```

Comment:
Maybe safer and more consistent to use JOB_BARRIER_TIMEOUT here... it'll make the test take longer, but less likely for TravisCI to choke.